### PR TITLE
[Infinity-1012] parallelize pr tests

### PR DESCRIPTION
**Pull request by jsrodman**
Goals:
 * run tests faster in parallel
 * Better logging (timestamps!)
 * Render failures unambiguously via junit xml (not done yet)
 * Try to reduce the number of environment variables floating around.

### CODE REVIEW

**Comment by susanxhuynh** [comment]
Looks good to me.

File: test.py [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+            help="requires a prior build")
+    parser.add_argument("--build-only", action='store_false', dest='run_tests')
+    parser.add_argument("--parallel", action='store_true',
+            help="Use more than one cluster to try to speed up the tests")
+    parser.add_argument("--cluster-count", type=int, default=2,
+            help="Ignored unless using --parallel.  "
+                 "Number of clusters to run tests on. "
+                 "Large values are likely to break CCM.")
+    parser.add_argument("--order", choices=('random', 'ordered'), default='random',
+            help="Run tests in randomr order, or the order given on the command line.  "
```
Comment (loren) [inline test.py]: typo nit: randomr

File: tools/fwinfo.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+def init_repo_root(repo_root):
+    global _repo_root
+    _repo_root=repo_root
+
+def add_framework(framework_name, repo_root=None):
+    if not repo_root:
+        repo_root=_repo_root
+    logger.debug("add_framework(%s, repo_root=%s)", framework_name, repo_root)
+    if framework_name in _framework_infos:
+        raise Exceptoin("Tried to add framework %s when it already exists")
```
Comment (loren) [inline tools/fwinfo.py]: Exception

**Comment by jsrodman** [comment]
BTW @loren , definitely fixing those things you pointed out; just merged test.py so arthur could use it.

File: test.py (hunk 1 above)
Comment (jsrodman) [inline test.py]: the ordering isn't implemented here, the full args were more or less trying to plan what I needed to support, and the pull request was kind of a minimum "this runs at all, and please stop me if this is awful"

**Pull request merged**